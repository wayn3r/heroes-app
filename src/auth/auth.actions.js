import { types } from '../types/types'

/**
 *
 * @param {String} name
 */
export const login = name => ({
    type: types.login,
    payload: { name },
})
export const logout = () => ({
    type: types.logout,
})
