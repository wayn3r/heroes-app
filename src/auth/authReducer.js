import { types } from '../types/types'

export const initialState = {
    name: '',
    logged: false,
}

export const authReducer = (state = initialState, action = {}) => {
    const { type, payload } = action
    switch (type) {
        case types.login:
            const { name } = payload
            return { name, logged: true }
        case types.logout:
            return { ...initialState }
        default:
            return state
    }
}
