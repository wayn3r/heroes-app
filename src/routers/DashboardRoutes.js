import { Navigate, Route, Routes } from 'react-router-dom'
import HeroPage from '../components/pages/HeroPage'
import NavBar from '../components/organisms/NavBar'
import DCPage from '../components/pages/DCPage'
import MarvelPage from '../components/pages/MarvelPage'
import SearchPage from '../components/pages/SearchPage'

const DashboardRoutes = () => {
    return (
        <>
            <NavBar />
            <div className="container mt-3">
                <Routes>
                    <Route path="marvel" element={<MarvelPage />} />
                    <Route path="dc" element={<DCPage />} />
                    <Route path="hero/:id" element={<HeroPage />} />
                    <Route path="search" element={<SearchPage />} />
                    <Route path="/" element={<Navigate to="marvel" />} />
                </Routes>
            </div>
        </>
    )
}

export default DashboardRoutes
