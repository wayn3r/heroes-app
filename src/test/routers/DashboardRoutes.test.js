import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'
import DashboardRoutes from '../../routers/DashboardRoutes'
import { loggedUser } from '../fixins/AuthContextValue'

describe('Pruebas en <DashboardRoutes />', () => {
    test('debe mostrarse correctamente', () => {
        const wrapper = mount(
            <AuthContext.Provider value={{ user: loggedUser }}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>,
        )
        expect(wrapper).toMatchSnapshot()
    })
})
