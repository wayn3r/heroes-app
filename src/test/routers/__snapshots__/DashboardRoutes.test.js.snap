// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes /> debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark px-5"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Wayner
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Marvel Page
      </h1>
      <hr />
      <div
        className="heroes-list animate__animated animate__fadeIn"
      >
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Spider Man"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Spider Man
                </div>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  Amazing Fantasy #15
                </p>
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Captain America"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Captain America
                </div>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                  Captain America Comics #1
                </p>
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Iron Man"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Iron Man
                </div>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                  Tales of Suspense #39
                </p>
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Thor"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Thor
                </div>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                  Journey into Myster #83
                </p>
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hulk"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Hulk
                </div>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                  The Incredible Hulk #1
                </p>
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Wolverine"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Wolverine
                </div>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                  The Incredible Hulk #180
                </p>
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Daredevil"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Daredevil
                </div>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                  Daredevil #1
                </p>
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hawkeye"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Hawkeye
                </div>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                  Tales of Suspense #57
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Cyclops"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Cyclops
                </div>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                  X-Men #1
                </p>
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card heroes-list__item"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Silver Surfer"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Silver Surfer
                </div>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                  The Fantastic Four #48
                </p>
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  Más detalles...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
